Anlab.Mvc/Services/DocumentSigningService.cs

```diff
@@ -0,0 +1,254 @@
+
+        return _eSignatureSettings.AuthorizationEndpoint + "?response_type=code" +
+               "&scope=" + scopes +
+               "&client_id=" + _eSignatureSettings.ClientId +
+               "&redirect_uri=" + path + "admin";
+    }
+
+    private async Task<EnvelopeDefinition> MakeEnvelope(Order order)
+    {
+        Debug.Assert(_httpContextAccessor.HttpContext != null, "_httpContextAccessor.HttpContext != null");
+
+        var cookies = _httpContextAccessor.HttpContext.Request.Cookies;
+
+        // add html doc and add signer tag to it
+        var env = new EnvelopeDefinition
+        {
+            EmailSubject = "Please sign this document",
+            // the document should expire after 2 days with no reminder
+            Notification = new Notification
+            {
+                UseAccountDefaults = "false",
+                Reminders = new Reminders
+                {
+                    ReminderEnabled = "false",
+                },
+                Expirations = new Expirations
+                {
+                    ExpireEnabled = "true",
+                    ExpireAfter = "2",
+                    ExpireWarn = "0"
+                }
+            }
+        };
+
+        // read html from url /order/document/{id}
+        using var handler = new HttpClientHandler { UseCookies = false };
+        using var client = new HttpClient(handler);
+        client.BaseAddress = new Uri(_eSignatureSettings.ApplicationBaseUri);
+
+        // add all cookies to the request
+        foreach (var cookie in cookies)
+        {
+            client.DefaultRequestHeaders.Add("Cookie", cookie.Key + "=" + cookie.Value);
+        }
+
+        var html = await client.GetStringAsync("/order/document/" + order.Id);
+
+        // add a document to the envelope
+        var doc = new Document
+        {
+            DocumentBase64 = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(html)),
+            Name = "Order acknowledgement",
+            FileExtension = "html",
+            DocumentId = "1"
+        };
+
+        env.Documents = new List<Document> {doc};
+
+        // add a signer to the envelope
+        var signer = new Signer
+        {
+            Email = order.Creator.Email,
+            Name = order.Creator.Name,
+            ClientUserId = SignerClientId,
+            RecipientId = "1",
+        };
+
+        // create sign here tab (signing field on document) based off of anchor string
+        var signHere = new SignHere
+        {
+            AnchorString = "**signature_1**",
+            AnchorUnits = "pixels",
+            // AnchorYOffset = "10",
+            AnchorXOffset = "20",
+        };
+
+        var dateSigned = new DateSigned
+        {
+            AnchorString = "**date_1**",
+            AnchorUnits = "pixels",
+            // AnchorYOffset = "10",
+            AnchorXOffset = "20",
+        };

```

Comment:
Should the signature location somehow be made configurable/dynamic? Or is this good enough for the few documents it's used on?